parser/src/main/kotlin/com/linkedin/dex/parser/DexParser.kt

```diff
@@ -84,14 +84,35 @@ private class DexParserCommand : CliktCommand() {
         fun readDexFiles(path: String): List<DexFile> {
             ZipInputStream(FileInputStream(File(path))).use { zip ->
                 return zip.entries()
-                        .filter { it.name.endsWith(".dex") }
-                        .map { zip.readBytes() }
+                        .filter {
+                            it.name.endsWith(".dex") or
+                                    (it.name.contains("secondary-program-dex-jars")
+                                            and it.name.endsWith(".jar"))
+                        }
+                        .map {
+                            if (it.name.endsWith(".jar")) {
+                                readSecondaryDex(zip.readBytes())
+                            } else {
+                                zip.readBytes()
+                            }
+                        }
                         .map { ByteBuffer.wrap(it) }
                         .map(::DexFile)
                         .toList()
             }
         }
 
+        private fun readSecondaryDex(jardex: ByteArray): ByteArray {
+            ZipInputStream(jardex.inputStream()).use { zip ->
+                return zip.entries()
+                    .filter {
+                        it.name.endsWith(".dex")
+                    }
+                    .map{ zip.readBytes() }
+                    .first()
```

Comment:
I'm confused why this is only the first - isn't the goal to supoprt multidex by putting the extra dex filess inside this jar?